kymatio/scattering1d/frontend/numpy_frontend.py

```diff
@@ -18,36 +18,24 @@ def __init__(self, J, shape, Q=1, T=None, max_order=2, average=True,
     def scattering(self, x):
         ScatteringBase1D._check_runtime_args(self)
         ScatteringBase1D._check_input(self, x)
-
-        batch_shape = x.shape[:-1]
-        signal_shape = x.shape[-1:]
-
-        x = x.reshape((-1, 1) + signal_shape)
+        x_shape = self.backend.shape(x)
+        batch_shape, signal_shape = x_shape[:-1], x_shape[-1:]
```

Comment:
what if i the input shape is `(batch, channels, signal_shape)`? i feel that this is dangerous, particularly given that `_check_input` is weak and even has a comment to say so: ttps://github.com/lostanlen/kymatio/blob/571e1b3d98c1e8e6ad8bb73a8cd81065bdbd99ec/kymatio/scattering1d/frontend/base_frontend.py#L207

maybe we should strengthen `_check_input`

Reply:
or maybe i'm wrong?

Reply:
although this was already part of the implementation